tensorflow/compiler/mlir/tosa/transforms/legalize_tfl.cc

```diff
@@ -4615,6 +4619,18 @@ void populateLegalizeTFLPatterns(MLIRContext* ctx,
   DEF_PATTERN_INSERT(TFLReal);
   DEF_PATTERN_INSERT(TFLImag);
   DEF_PATTERN_INSERT(TFLRFFT2d);
+#undef DEF_PATTERN_INSERT
+}
+
+void populateLegalizeTFLToTensorPatterns(MLIRContext* ctx,
+                                         RewritePatternSet& patterns) {
+#define DEF_PATTERN_INSERT(PAT)                                   \
+  patterns.addWithLabel<Convert##PAT##Op>({#PAT"ToTensor"}, ctx, \
+                                          /*tosaOnly*/ false);
+
+  DEF_PATTERN_INSERT(TFLGather);
```

Comment:
Given this still involve generating `tosa` operations, let's just including it with the basic `TFLToTosaPatterns`, just include it as a separate group with a comment. If it only used `tensor` operations (e.g. `tfl.shape`) then it would make sense separately, but given its use if `tosa.gather` there are not as much benefits.

We can assume that if a user does not want `tensor.dim` but it exists we can just inform them there is no matching `tosa` operation yet.